stl/inc/valarray

```diff
@@ -548,11 +548,11 @@ private:
 
     void _Tidy_deallocate() noexcept {
         if (_Myptr) { // destroy elements
-            for (size_t _Idx = 1; _Idx < _Mysize; ++_Idx) {
+            for (size_t _Idx = 0; _Idx < _Mysize; ++_Idx) {
                 _Destroy_in_place(_Myptr[_Idx]);
             }
 
-            delete _Myptr;
+            ::operator delete(static_cast<void*>(_Myptr));

```

Comment:
I believe that this is introducing a different kind of mismatch for highly-aligned types. This is paired with:

```cpp
template <class _Ty>
_Ty* _Allocate_for_op_delete(size_t _Count) {
    // allocates space for _Count objects of type _Ty
    if (_Count == 0) {
        return nullptr;
    }

    const size_t _Bytes = _Get_size_of_n<sizeof(_Ty)>(_Count);
#ifdef __cpp_aligned_new
    constexpr bool _Extended_alignment = alignof(_Ty) > __STDCPP_DEFAULT_NEW_ALIGNMENT__;
    if _CONSTEXPR_IF (_Extended_alignment) {
        return static_cast<_Ty*>(::operator new (_Bytes, align_val_t{alignof(_Ty)}));
    } else
#endif // __cpp_aligned_new
    {
        return static_cast<_Ty*>(::operator new(_Bytes));
    }
}
```

Previously, `delete _Myptr` would select an `operator delete` overload taking `align_val_t` if available (WG21-N4868 [expr.delete]/10.2). Now, even when `_Allocate_for_op_delete` selects `::operator new (_Bytes, align_val_t{alignof(_Ty)})`, plain `::operator delete(static_cast<void*>(_Myptr))` is called here. This appears to mismatch `_aligned_malloc` with `free`.

To fix this, I believe we need the same logic that `_Allocate_for_op_delete` is using.